metadata-ingestion/tests/unit/test_protobuf_util.py

```diff
@@ -0,0 +1,302 @@
+import pytest
+
+from datahub.ingestion.extractor.protobuf_util import (
+    ProtobufSchema,
+    protobuf_schema_to_mce_fields,
+)
+from datahub.metadata.schema_classes import ArrayTypeClass
+
+SCHEMA_WITH_SINGLE_MESSAGE_EMPTY_MESSAGE = """
+syntax = "proto3";
+
+message Test1 {
+}
+"""
+
+SCHEMA_WITH_SINGLE_MESSAGE_SINGLE_FIELD = """
+syntax = "proto3";
+
+message Test2 {
+    string field_1 = 1;
+}
+"""
+
+SCHEMA_WITH_TWO_MESSAGES_ENUM = """
+syntax = "proto3";
+
+message Test3 {
+    string field_1 = 1;
+
+    enum anEnum {
+        first = 0;
+        second = 1;
+    }
+
+    anEnum field_2 = 2;
+}
+
+message Test4 {
+    int64 anInteger = 1;
+}
+"""
+
+SCHEMA_NESTED = """
+syntax = "proto3";
+
+message Test5 {
+    Nested1 f1 = 1;
+    message Nested1 {
+        Nested2 f2 = 1;
+        message Nested2 {
+            Nested3 f3 = 1;
+            message Nested3 {
+                message Nested4 {}
+                Nested4 f4 = 1;
+            }
+        }
+    }
+}
+"""
+
+SCHEMA_REPEATED = """
+syntax = "proto3";
+
+message Test6 {
+    repeated int64 aList = 1;
+}
+"""
+
+SCHEMA_REPEATED_WITH_MESSAGE_NESTED_TYPE = """
+syntax = "proto3";
+
+message Test7 {
+    repeated Nested aList = 1;
+
+    message Nested {
+        string name = 1;
+    }
+}
+"""
+
+SCHEMA_MAP = """
+syntax = "proto3";
+
+message Test8 {
+    map<string, int64> map_1 = 1;
+    map<string, Nested> map_2 = 2;
+
+    message Nested {
+        string aString = 1;
+    }
+}
+"""
+
+SCHEMA_COMPLEX = """
+syntax = "proto3";
+
+message Test9 {
+    string string_field_1  = 1;
+    bool   boolean_field_1 = 2;
+    int64  int64_field_1   = 3;
+
+    EmptyNested emptyMsg = 4;
+
+    // an empty message
+    message EmptyNested {};
+
+    oneof payload {
+        string pl_1 = 5;
+        int64  pl_2 = 6;
+    }
+
+    enum anEnum {
+        idle = 0;
+        spinning = 1;
+    }
+}
+
+message Test10 {
+    int64 an_int_64_field = 1;
+}
+"""
+
+SCHEMA_WITH_RECURSIVE_MESSAGE = """
+syntax = "proto3";
+
+message Test11 {
+    string a_property = 1;
+    Test11 recursive = 2;
+}
+"""
+
+def test_protobuf_schema_to_mce_fields_with_single_empty_message():
+    schema = SCHEMA_WITH_SINGLE_MESSAGE_EMPTY_MESSAGE
+    fields = protobuf_schema_to_mce_fields(ProtobufSchema("main_1.proto", schema))
+
+    assert 0 == len(fields)
+
+
+def test_protobuf_schema_to_mce_fields_with_single_message_single_field():
+    schema = SCHEMA_WITH_SINGLE_MESSAGE_SINGLE_FIELD
+    fields = protobuf_schema_to_mce_fields(ProtobufSchema("main_2.proto", schema))
```

Comment:
I found that return of protobuf_schema_to_mce_field is List[SchemaField]. May I know why you use this rather than mce field?

Reply:
Got it, thanks